### PR TITLE
[WIP] Update contour version to 1.2.1

### PR DESCRIPTION
**Pull request by shashwathi**
In this PR we have attempted to update to contour 1.2.1 version. 

Note: 
- we have edited factory and lister files for testing purpose.
- No inject, listers, clientset, informers are generated for contour APIs.

Help: If @mattmoor and @n3wscott have any ideas on how we can generate factory and lister go files directly that would be great.

cc @andrew-su 

/assign @mattmoor 
/assign @n3wscott

### CODE REVIEW

**Comment by andrew-su** [comment]
@googlebot I consent.

File: pkg/reconciler/contour/resources/httpproxy.go [hunk 1]
```diff
@@ -154,6 +154,10 @@ func MakeHTTPProxies(ctx context.Context, ing *v1alpha1.Ingress, serviceToProtoc
 		}
 
 		base := v1.HTTPProxy{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "HTTPProxy",
```
Comment (mattmoor) [inline pkg/reconciler/contour/resources/httpproxy.go]: Format Go code:
```suggestion
				Kind:       "HTTPProxy",
```

**Comment by mattmoor** [comment]
Just saw Contour 1.3 cut, so we're getting pretty far behind.  Has anyone had a chance to discuss restoring the `// +genclient` upstream, or simply carrying a patch for that to unblock generating the client?

I'd probably start with the latter, so we know what change we want upstream (and to unblock) and then starting the convo for the former.

**Comment by shashwathi** [comment]
closing in favor #79

**Pull request closed without merging**